From af252e4a95c05d4e17678eaa58b411513d60ffe2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 29 Mar 2025 04:07:33 +0100
Subject: [PATCH] Raise existing window instead of opening new one on
 activation

Glade does not really support multiple main windows, e.g. closing one
closes all windows, so it's better to avoid it.
---
 src/main.c | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/src/main.c b/src/main.c
index 019c89f5..63da8b06 100644
--- a/src/main.c
+++ b/src/main.c
@@ -101,13 +101,22 @@ static void
 activate (GApplication *application)
 
 {
-  GladeWindow *window = GLADE_WINDOW (glade_window_new ());
+  GtkApplication *app = GTK_APPLICATION (application);
+  GtkWindow *window;
+
+  if ((window = gtk_application_get_active_window (app)))
+    {
+      gtk_window_present (window);
+      return;
+    }
+
+  window = GTK_WINDOW (glade_window_new ());
 
   gtk_application_add_window (GTK_APPLICATION (application),
-                              GTK_WINDOW (window));
+                              window);
 
   if (without_devhelp == FALSE)
-    glade_window_check_devhelp (window);
+    glade_window_check_devhelp (GLADE_WINDOW (window));
 
   gtk_widget_show (GTK_WIDGET (window));
 }
-- 
2.49.0

